libs/patchwheel/parse_test.go

```diff
@@ -0,0 +1,124 @@
+package patchwheel
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestCalculateNewVersion(t *testing.T) {
+	tests := []struct {
+		name             string
+		info             *WheelInfo
+		mtime            time.Time
+		expectedVersion  string
+		expectedFilename string
+	}{
+		{
+			name: "basic version",
+			info: &WheelInfo{
+				Distribution: "mypkg",
+				Version:      "1.2.3",
+				Tags:         []string{"py3", "none", "any"},
+			},
+			mtime:            time.Date(2025, 3, 4, 12, 34, 56, 780_000_000, time.UTC),
+			expectedVersion:  "1.2.3+2025030412345678",
+			expectedFilename: "mypkg-1.2.3+2025030412345678-py3-none-any.whl",
+		},
+		{
+			name: "existing plus version",
+			info: &WheelInfo{
+				Distribution: "mypkg",
+				Version:      "1.2.3+local",
+				Tags:         []string{"py3", "none", "any"},
+			},
+			mtime:            time.Date(2025, 3, 4, 12, 34, 56, 100_000_000, time.UTC),
+			expectedVersion:  "1.2.3+2025030412345610",
+			expectedFilename: "mypkg-1.2.3+2025030412345610-py3-none-any.whl",
+		},
+		{
+			name: "complex distribution name",
+			info: &WheelInfo{
+				Distribution: "my-pkg-name",
+				Version:      "1.2.3",
+				Tags:         []string{"py3", "none", "any"},
+			},
+			mtime:            time.Date(2025, 3, 4, 12, 34, 56, 0, time.UTC),
+			expectedVersion:  "1.2.3+2025030412345600",
+			expectedFilename: "my-pkg-name-1.2.3+2025030412345600-py3-none-any.whl",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			newVersion, newFilename := CalculateNewVersion(tt.info, tt.mtime)
+			if newVersion != tt.expectedVersion {
+				t.Errorf("expected version %s, got %s", tt.expectedVersion, newVersion)
+			}
+			if newFilename != tt.expectedFilename {
+				t.Errorf("expected filename %s, got %s", tt.expectedFilename, newFilename)
+			}
+		})
+	}
+}
+
+func TestParseWheelFilename(t *testing.T) {
+	tests := []struct {
```

Comment:
Not sure if it matters but this is an incorrect name but from what I see it will pass, shall we handle this?

```
myproj-foobar-py3-none-any.whl
```

Reply:
I'm not sure I want to add a lot of validations - in the end, we're only parsing for the purpose of patching, not to give user diagnostics here.

Adding validations comes with a risk of accidentally rejecting valid inputs.

Reply:
It does have some `Wheel-Version: 1.9` in the example of the document, not sure what the effect of it though, any ideas?

Reply:
> Wheel-Version is the version number of the Wheel specification.

https://peps.python.org/pep-0427/